contrib/hashicorp/vault/vault_test.go

```diff
@@ -0,0 +1,404 @@
+		assert.Equal(http.MethodPut+" "+fullPath, span.Tag(ext.ResourceName))
+		assert.Equal(ext.SpanTypeHTTP, span.Tag(ext.SpanType))
+		assert.Equal(200, span.Tag(ext.HTTPCode))
+		assert.Nil(span.Tag(ext.Error))
+		assert.Nil(span.Tag(ext.ErrorMsg))
+		assert.Equal(namespace, span.Tag("vault.namespace"))
+	})
+
+	t.Run("read", func(t *testing.T) {
+		assert := assert.New(t)
+		mt := mocktracer.Start()
+		defer mt.Stop()
+
+		// Write key with namespace first
+		data := map[string]interface{}{"Key1": "Val1", "Key2": "Val2"}
+		_, err = client.Logical().Write(key, data)
+		if err != nil {
+			t.Fatal(err)
+		}
+		// Read key with namespace
+		_, err = client.Logical().Read(key)
+		if err != nil {
+			t.Fatal(err)
+		}
+		spans := mt.FinishedSpans()
+		assert.Len(spans, 2)
+		span := spans[1]
+
+		assert.Equal("vault", span.Tag(ext.ServiceName))
+		assert.Equal(fullPath, span.Tag(ext.HTTPURL))
+		assert.Equal(http.MethodGet, span.Tag(ext.HTTPMethod))
+		assert.Equal(http.MethodGet+" "+fullPath, span.Tag(ext.ResourceName))
+		assert.Equal(ext.SpanTypeHTTP, span.Tag(ext.SpanType))
+		assert.Equal(200, span.Tag(ext.HTTPCode))
+		assert.Nil(span.Tag(ext.Error))
+		assert.Nil(span.Tag(ext.ErrorMsg))
+		assert.Equal(namespace, span.Tag("vault.namespace"))
+	})
+}
+
+func TestOption(t *testing.T) {
+	ts, cleanup := setupServer(t)
+	defer cleanup()
+
+	for ttName, tt := range map[string]struct {
+		opts []Option
+		test func(assert *assert.Assertions, span mocktracer.Span)
+	}{
+		"DefaultOptions": {
+			opts: []Option{},
+			test: func(assert *assert.Assertions, span mocktracer.Span) {
+				assert.Equal(defaultServiceName, span.Tag(ext.ServiceName))
+				assert.Nil(span.Tag(ext.EventSampleRate))
+			},
+		},
+		"CustomServiceName": {
+			opts: []Option{WithServiceName("someServiceName")},
+			test: func(assert *assert.Assertions, span mocktracer.Span) {
+				assert.Equal("someServiceName", span.Tag(ext.ServiceName))
+			},
+		},
+		"WithAnalyticsTrue": {
+			opts: []Option{WithAnalytics(true)},
+			test: func(assert *assert.Assertions, span mocktracer.Span) {
+				assert.Equal(1.0, span.Tag(ext.EventSampleRate))
+			},
+		},
+		"WithAnalyticsFalse": {
+			opts: []Option{WithAnalytics(false)},
+			test: func(assert *assert.Assertions, span mocktracer.Span) {
+				assert.Nil(span.Tag(ext.EventSampleRate))
+			},
+		},
+		"WithAnalyticsLastOptionWins": {
+			opts: []Option{WithAnalyticsRate(0.7), WithAnalytics(true)},
+			test: func(assert *assert.Assertions, span mocktracer.Span) {
+				assert.Equal(1.0, span.Tag(ext.EventSampleRate))
+			},
+		},
+		"WithAnalyticsRateNegative": {
+			opts: []Option{WithAnalyticsRate(-10.0)},
+			test: func(assert *assert.Assertions, span mocktracer.Span) {
+				assert.Nil(span.Tag(ext.EventSampleRate))
+			},
+		},
+		"WithAnalyticsRateGreaterThanOne": {
+			opts: []Option{WithAnalyticsRate(10.0)},
+			test: func(assert *assert.Assertions, span mocktracer.Span) {
+				assert.Nil(span.Tag(ext.EventSampleRate))
+			},
+		},
```

Comment:
What you are testing here is outside the scope of this package. We can have them if you like, but then let's add them in the right place as a counterpart to [this file](https://github.com/DataDog/dd-trace-go/blob/v1/contrib/net/http/option.go#L53-L61).

Reply:
That's a good point. I'll remove those tests. That's not behavior this package is responsible for.